ironfish/src/primitives/target.ts

```diff
@@ -104,26 +103,35 @@ export class Target {
    * @param previousBlockTarget the block's previous block header's target
    */
   static calculateDifficulty(
+    consensus: Consensus,
+    sequence: number,
     time: Date,
     previousBlockTimestamp: Date,
     previousBlockDifficulty: bigint,
-    targetBlockTimeInSeconds: number,
-    targetBucketTimeInSeconds: number,
-    maxBuckets: number,
   ): bigint {
+    const targetBlockTime = consensus.parameters.targetBlockTimeInSeconds
+    const targetBucketTime = consensus.parameters.targetBucketTimeInSeconds
+    const enableFishHashSequence = consensus.parameters.enableFishHash
+    const maxBuckets = consensus.getDifficultyBucketMax(sequence)
+
     const diffInSeconds = (time.getTime() - previousBlockTimestamp.getTime()) / 1000
 
     let bucket = Math.floor(
-      (diffInSeconds - targetBlockTimeInSeconds + Math.floor(targetBucketTimeInSeconds / 2)) /
-        targetBucketTimeInSeconds,
+      (diffInSeconds - targetBlockTime + Math.floor(targetBucketTime / 2)) / targetBucketTime,
     )
 
     // Should not change difficulty by more than `maxBuckets` buckets from last block's difficulty
     bucket = Math.min(bucket, maxBuckets)
 
-    const difficulty =
+    let difficulty =
       previousBlockDifficulty - (previousBlockDifficulty / 2048n) * BigInt(bucket)
 
+    // A one-time difficulty adjustment when the mining algorithm changes from
+    // blake3 to fishhash, since the fishhash hashrate is much lower than blake3
+    if (sequence === enableFishHashSequence) {
+      difficulty /= 100n
```

Comment:
Should maybe have this in a constant. Was also thinking about if we want this number in the consensus params too. Just in case we wanted to make it different on different networks. Maybe a consensus param called `oneTimeFishHashAdjustment` or something like that